### ISSUE TITLE
OpenEmory/Python3 Deployment

### ISSUE THREAD

**Issue by rotated8** [emory-libraries/OpenEmory#24]
These tasks are not blocked by others and can be started today.
- [x] @jcrompton42 - Check the sizes of the QA and Prod servers
- [ ] @alexBLR @kbowaterskelly @jcrompton42 - Set up a new test instance on the QA server.
- [ ] @kbowaterskelly - Inform DNS team of impending changes.

These tasks are the critical path to getting the updated application into production, and need to happen in order.
- [ ] @alexBLR - Contact SCO about changing the QA environment to prod Fedora and database.
- [ ] @alexBLR - With SCO's ok, make the change to prod Fedora and database.
- [ ] @alexBLR - Test QA application with Prod connections
- [ ] @alexBLR - Get SCO's permission and confirm a date to switch the updated QA application to Prod.
- [ ] @kbowaterskelly - Inform DNS team of confirmed date.
- [ ] @jcrompton42 @kbowaterskelly - Complete any Apache changes to move QA to prod

**Comment by rotated8** [comment]
@jcrompton42 reports the difference between QA and Prod are two CPU cores.

**Comment by jcrompton42** [comment]
waiting on go ahead from @alexBLR to proceed with next steps.

**Comment by alexBLR** [comment]
Added new API Key for Sherpa Romeo to 1Password

**Comment by jcrompton42** [comment]
will take a look at the fedora/wlibqas001 connection tomorrow

**Comment by jcrompton42** [comment]
Wlibqas001 is unable to reach libmysqlprod1 on port 3306. I will open a firewall ticket to get our network team to grant this access.

**Comment by kbowaterskelly** [comment]
I ordered an SSL cert in preparation for the transition. A scheduled change for the DNS may or may not be necessary but please assess impact and let me know

**Comment by alexBLR** [comment]
@jcrompton42 fixed the issue with the 401 connection error to prod DB. I am testing out the prod environment again. Then I will email SCO so they can do some testing on their end.

**Comment by alexBLR** [comment]
I changed to prod environment again and somehow I am getting the error about connecting to wlibqas001 DB even though we are connectinng to it anywhere: "django.db.utils.OperationalError: (1045, "Access denied for user 'openemory'@'wlibqas001.library.emory.edu' (using password: YES)")"

**Comment by alexBLR** [comment]
@jcrompton42 it looks like the issue with the Prod db still persists. When I switched to libmysqldev1.cc.emory.edu the error above is gone.

**Comment by jcrompton42** [comment]
@alexBLR because of the way mysql does authentication I believe openemory@'wlibqas001' is actually a different user from openemory@'webprd002' so I have opened a ticket for the DB team to grant access.

**Comment by jcrompton42** [comment]
@alexBLR openemory can now connect to libmysqlprod1's db.

**Comment by jcrompton42** [comment]
Fixed, the error_log for httpd was being missed by the logrotate rules and was at 9G, rotated and removed, so you should have the space now.

**Comment by jcrompton42** [comment]
@alexBLR nah, different issue, dns hasn't changed, but it is fixed now!

**Comment by kbowaterskelly** [comment]
I am waiting for a date to schedule DNS changes

**Comment by alexBLR** [comment]
It looks like everything works in Open emory except for LDAP. @jcrompton42  and I discussed removing VPN access to the qa box and enabling ldap permissions as well.

**Comment by jcrompton42** [comment]
I have submitted a ticket for LDAP auth for testopen

**Comment by alexBLR** [comment]
LDAP issue has been been fixed. We are waiting for VPN access to be disabled on the server.

**Comment by jcrompton42** [comment]
Elliott has a list of configuration changes he wants on apache before allowing public access. I will be working on implementing those tomorrow.

**Comment by jcrompton42** [comment]
Firewall change are scheduled to be put in place over the weekend.

**Comment by alexBLR** [comment]
I emailed Jody and Kylie to let them know that they will be able to do another round of testing by the end of this week, when the firewall is off.

**Comment by jcrompton42** [comment]
I've put in another ticket, and supposedly the firewall exception will be put in tonight for realz.
